packages/application-generic/src/services/bull-mq/bull-mq.service.ts

```diff
@@ -46,35 +43,18 @@ export {
 export class BullMqService {
   private _queue: Queue;
   private _worker: Worker;
-  private inMemoryProviderService: InMemoryProviderService;
+  private workflowInMemoryProviderService: WorkflowInMemoryProviderService;
 
   public static readonly pro: boolean =
     process.env.NOVU_MANAGED_SERVICE !== undefined;
 
   constructor() {
-    this.inMemoryProviderService = new InMemoryProviderService(
-      this.selectProvider()
-    );
-  }
-
-  /**
-   * Rules for the provider selection:
-   * - For our self hosted users we assume all of them have a single node Redis
-   * instance.
-   * - For Novu we will use MemoryDB. We fallback to a Redis Cluster configuration
-   * if MemoryDB not configured properly. That's happening in the provider
-   * mapping in the /in-memory-provider/providers/index.ts
-   */
-  private selectProvider(): InMemoryProviderEnum {
-    if (process.env.IS_DOCKER_HOSTED) {
-      return InMemoryProviderEnum.REDIS;
-    }
-
-    return InMemoryProviderEnum.MEMORY_DB;
+    this.workflowInMemoryProviderService =
```

Comment:
Now the BullMQ service will have no knowledge of the in-memory provider selected as it is not its responsibility.